lib/view/page/ideas_main_page.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+import 'package:idea_tracker/model/idea.dart';
+import 'package:idea_tracker/view/page/create_idea_page.dart';
+import 'package:idea_tracker/view/page/idea_details_page.dart';
+
+class IdeasMainPage extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.white.withOpacity(0.9),
+      appBar: AppBar(
+        title: Text('Ideas'),
+      ),
+      floatingActionButton: FloatingActionButton(
+        onPressed: () {
+          Navigator.push(
+              context,
+              MaterialPageRoute(
+                  builder: (context) => CreateIdeaPage('Create Idea')));
+        },
+        child: Icon(Icons.add),
+      ),
+      body: ListView.builder(
+          itemCount: ideas.length,
+          itemBuilder: (context, index) {
+            return IdeaCard(
+              idea: ideas[index],
+            );
+          }),
+    );
+  }
+}
+
+final List<Idea> ideas = [
```

Comment:
The ideas should be retrieved by an IdeasMainPageController. The controller retrieves ideas from the IdeaService.

Reply:
> The ideas should be retrieved by an IdeasMainPageController. The controller retrieves ideas from the IdeaService.

I think I'll need more help with this one. I've been looking at IdeaService.dart and the other page controllers created by others but I can't seem to wrap my head around them.

Reply:
If you call locator<IdeaService>().getAll() it will return the list of ideas you will be working with.

This is a simple example of how to use a controller: https://pastebin.com/AsXu1Wth

You would be using the IdeaService in the onControllerReady function.

Let me know if you need more info

Reply:
![image](https://user-images.githubusercontent.com/60046603/105633072-68953100-5e4e-11eb-99c6-1df23086849e.png)

![image](https://user-images.githubusercontent.com/60046603/105633074-6af78b00-5e4e-11eb-9b1c-8a13d48c16f3.png)

@DanielQuick please see attached images for my code. I think I'm calling the locator().getAll at the wrong place

Reply:
@labody A couple things I see.

1. The IdeasMainPage should not have the `final List<Idea> ideas;`
2. The IdeasMainPage should be requesting the controller to get the ideas, not set them itself.
3. You are fetching the IdeaService incorrectly.

How I would do it:

IdeasMainPage
```
onControllerReady: (controller) {
    controller.initialize();
},

// In the scaffold
body: ListView.builder(
    itemCount: controller.ideas.length,
    itemBuilder: (context, index) {
        return IdeaCard(
            idea: controller.ideas[index],
        );
    }
),
```

IdeasMainPageController
```
List<Idea> _ideas = [];
List<Idea> get ideas => _ideas;
set ideas (List<Idea> value) {
    if (_ideas == value) return;

    _ideas = value;
    notifyListeners();
}

initialize() async {
    ideas = await locator<IdeaService>().getAll();
}
```

Reply:
Thanks for the breakdown @DanielQuick Code's working now.

Quick question, what's the purpose of notifyListeners() ?

Reply:
In this case, notifyListeners causes the view to be rebuilt.

It's part of the Provider package.